src/workflows-app/SubmissionHistory.test.js

```diff
@@ -0,0 +1,561 @@
+import { act, render, screen, waitFor, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import _ from 'lodash/fp';
+import { div, h } from 'react-hyperscript-helpers';
+import selectEvent from 'react-select-event';
+import { MenuTrigger } from 'src/components/PopupTrigger';
+import { Ajax } from 'src/libs/ajax';
+import { getConfig } from 'src/libs/config';
+import { BaseSubmissionHistory } from 'src/workflows-app/SubmissionHistory';
+import { mockAbortResponse, mockAzureApps, mockAzureWorkspace } from 'src/workflows-app/utils/mock-responses';
+
+// Necessary to mock the AJAX module.
+jest.mock('src/libs/ajax');
+jest.mock('src/libs/notifications.js');
+jest.mock('src/libs/ajax/leonardo/Apps');
+jest.mock('src/libs/nav', () => ({
+  getCurrentUrl: jest.fn().mockReturnValue(new URL('https://app.terra.bio')),
+  getLink: jest.fn(),
+  goToPath: jest.fn(),
+}));
+jest.mock('src/components/PopupTrigger', () => {
+  const originalModule = jest.requireActual('src/components/PopupTrigger');
+  return {
+    ...originalModule,
+    MenuTrigger: jest.fn(),
+  };
+});
+jest.mock('src/libs/state', () => ({
+  ...jest.requireActual('src/libs/state'),
+  getUser: jest.fn(),
+}));
+// Mocking feature preview setup
+jest.mock('src/libs/feature-previews', () => ({
+  ...jest.requireActual('src/libs/feature-previews'),
+  isFeaturePreviewEnabled: jest.fn(),
+}));
+jest.mock('src/libs/config', () => ({
```

Comment:
`configStore` gets automatically mocked in test setup. This might be unnecessary.

Reply:
Ended up needing this for some mock urls used later in the code